File: C# Codewars/8kyu/Sum Mixed Array.cs
/*
Description

Given an array of integers as strings and numbers, return the sum of the array values as if all were numbers.

Return your answer as a number.
*/

using static System.Convert;
using System.Linq;

public class Kata
{
    public static int SumMix(object[] x)
    {
        return x.Sum(ToInt32);
    }
}

File: Typescript Codewars/8kyu/Sum Mixed Array.ts
/*
Description

Given an array of integers as strings and numbers, return the sum of the array values as if all were numbers.

Return your answer as a number.
*/

export function sumMix(x: any[]): number {
  let sum = 0;
  for (let i = 0; i < x.length; i++) {
    if (typeof x[i] === "string") {
      x[i] = Number(x[i]);
    }
    sum += x[i];
  }
  return sum;
}
